pymodbus/transport/transport_serial.py

```diff
@@ -38,7 +38,7 @@ def setup(self):
 
     def close(self, exc: Exception | None = None) -> None:
         """Close the transport gracefully."""
-        if not self.sync_serial:
+        if not hasattr(self, 'sync_serial'):
```

Comment:
see below please.

Reply:
Yes, I understand your concern.  Permitting `sync_serial` to be `None` means introducing more code to guard against calling it when `None`, so I found it cleanest to simply `del` the whole object.

Reply:
Or telling mypy to keep silent !!! we do not add code, just to please mypy.

Reply:
My experience is actually different a lot of the juniors do not understand nor agree with type checking at all....

There are many python developers who see the lack of type checking as a big advantage...being a C/C++/C# I am very used to strict type checking, but have also seen how it can make code a lot more complicated.

Personally I think the bridge (free use / type checking) python tries to build might end being a split into 2 products.

I like the freedom python allows, but see that mypy catches many problems....so for me a balance is the correct choice, meaning think if what mypy suggest mamés the code better if not ask mypy to stop complaining.